processor/processor.go

```diff
@@ -2566,10 +2566,13 @@ func (proc *Handle) transformSrcDest(
 	}
 }
 
-func (proc *Handle) saveDroppedJobs(failedJobs []*jobsdb.JobT, tx *jobsdb.Tx) error {
-	if len(failedJobs) > 0 {
+func (proc *Handle) saveDroppedJobs(droppedJobs []*jobsdb.JobT, tx *jobsdb.Tx) error {
+	if len(droppedJobs) > 0 {
+		for i := range droppedJobs { // each dropped job should have a unique jobID in the scope of the batch
+			droppedJobs[i].JobID = int64(i)
```

Comment:
Curious can't we use [e.Metadata.JobID](https://github.com/rudderlabs/rudder-server/blob/10c0ffe63c7d472112e3316bd05157b85a2dccc5/processor/transformer/transformer.go#L61) in [here](https://github.com/rudderlabs/rudder-server/blob/f5b8e7b4976945207ec0c011ed6d7fe30e1eb323/processor/events_response.go#L55)?

Reply:
Not really, processor operates on messageIDs